src/adapter/src/coord/dataflows.rs

```diff
@@ -772,7 +773,9 @@ fn eval_unmaterializable_func(
             pack(Datum::from(&*state.config().environment_id.to_string()))
         }
         UnmaterializableFunc::MzNow => match logical_time {
-            None => coord_bail!("cannot call mz_now in this context"),
+            // We aren't able to resolve MzNow here, so leave it as is. Eval functions later on will
+            // fail if they attempt to evaluate it.
+            None => Ok(MirScalarExpr::CallUnmaterializable(f.clone())),
             Some(logical_time) => pack(Datum::MzTimestamp(logical_time)),
```

Comment:
Perhaps the easiest way to support both the old and the new behavior would be to not use an `Option` for `logical_time` but instead a dedicated enum that allows to differentiate the caller's intentions. E.g:
```rust
enum EvalTime {
    Time(mz_repr::Timestamp),
    Deferred,  // skips `mz_now`
    NotAvailable,  // returns an error
}
```

Reply:
Done.